bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/discovery/ThingDiscoveryService.java

```diff
@@ -79,7 +79,8 @@ public class ThingDiscoveryService extends AbstractDiscoveryService implements T
             new AbstractMap.SimpleEntry<>("ROOM_CLIMATE_CONTROL", BoschSHCBindingConstants.THING_TYPE_CLIMATE_CONTROL),
             new AbstractMap.SimpleEntry<>("INTRUSION_DETECTION_SYSTEM", BoschSHCBindingConstants.THING_TYPE_INTRUSION_DETECTION_SYSTEM),
             new AbstractMap.SimpleEntry<>("HUE_LIGHT", BoschSHCBindingConstants.THING_TYPE_SMART_BULB),
-            new AbstractMap.SimpleEntry<>("WRC2", BoschSHCBindingConstants.THING_TYPE_WINDOW_CONTACT)
```

Comment:
According to my research `WRC2` is a universal switch, not a window contact. The device model for the window contact should be `SWD`. Please confirm.

Reply:
Who can confirm this? 🙂 If you are sure about this, I think we can merge.

Reply:
I think @GerdZanker should be able to confirm it. But I just found other associations that I would like to change, so I converted this pull request to a draft for now.

Reply:
My changes should be complete now. Open for review again.

Reply:
@GerdZanker should we associate the 2nd generation models with ones we have already? For example `SWD2` with the existing door/window contact implementation? The 2nd generation models normally have more features, but I think it should be possible to use the common features with our existing implementation in most cases. For example, I can already use the Door/Window Sensor II and the Thermostat II. Once we fully support the 2nd generation models, users would have to re-create their things. But this would also be the case if they go for what we currently support.

Reply:
> I think @GerdZanker should be able to confirm it. But I just found other associations that I would like to change, so I converted this pull request to a draft for now.

I can confirm that the window-contact and smart-plug work for me.

Reply:
I will throw in a few things to consider: It's not always desirable to map each model/revision of a device to different thing type. This might not be preferable if there are only slight differences, as it can easily get unmanageable. As an example, the Hunter Douglas binding uses one thing type for all different types of shades. We have this many models:

https://github.com/openhab/openhab-addons/blob/ce3f6733807324ff09e63c26b8f96cb1bcdecd70/bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java#L59-L99

with these different capabilities:

https://github.com/openhab/openhab-addons/blob/ce3f6733807324ff09e63c26b8f96cb1bcdecd70/bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java#L39-L57

We use the capabilities to dynamically modify the available channels to reflect the actual capabilities.

Similarly, we have only one bridge type since both PowerView Hub Gen 1 and Gen 2 expose almost the same API with only minor differences. With Gen 3 the API changed completely, so here a new bridge type is added to avoid too many mutually exclusive configuration options for Gen 1-2 vs. Gen 3 configuration.

The same is true for the Hue binding where Bridge 1 and Bridge 2 expose the same API. Bridge 2 now also exposes a new API v2, so for this we are introducing a new bridge type because it's completely different.

This is just to say that you might want to avoid a strict rule of 1:1 thing type -> device model relationship. If, for example, a new generation of a thermostat is very similar to the previous model and only adds a few features that could be exposed as additional channels, this might be worth considering. Dynamic channels will require either to be able to identify the model from the API, or requiring the user to configure the model explicitly.

There is also the Denon/Marantz binding which is compatible with probably almost 100 different receivers, where 5-10 new models are released every year. Luckily they are very similar when only some differences in available input sources, number of zones etc.

So in this description - https://www.openhab.org/docs/developer/bindings/#describing-things - a "device" could be interpreted as "thermostat" in your example, rather than "Thermostat II".

Reply:
Thank you for these very useful insights @jlaur. We will consider them carefully when we implement support for the 2nd generation devices and will add you as reviewer if that's ok with you. Thanks for your support :+1: